### ISSUE TITLE
URI exception for bitbucket

### ISSUE THREAD

**Issue by alanpurple** [git-for-windows/git#1194]
fatal: UriFormatException encountered.
   actualUrl

for VS2017, everything has failed

Error encountered while pushing to the remote repository: Git failed with a fatal error.
fatal: UriFormatException encountered.
   actualUrl
fatal: could not read Password for 'https://alanpurple@bitbucket.org': terminal prompts disabled
Pushing to https://alanpurple@bitbucket.org/alanpurple/advisordemo.git

for any shells( windows powershell or git bash ), same error occurred but I can enter password then, so the job could be accomplished

Windows 10 pro creator x64
git for windows 2.13.0
Visual studio 2017.2

**Comment by dscho** [comment]
> fatal: UriFormatException encountered.
> actualUrl
> 
> for VS2017, everything has failed
> 
> Error encountered while pushing to the remote repository: Git failed with a fatal error.
> fatal: UriFormatException encountered.
> actualUrl
> fatal: could not read Password for 'https://alanpurple@bitbucket.org': terminal prompts disabled
> Pushing to https://alanpurple@bitbucket.org/alanpurple/advisordemo.git
> 
> for any shells( windows powershell or git bash ), same error occurred but I can enter password then, so the job could be accomplished
> 
> Windows 10 pro creator x64
> git for windows 2.13.0
> Visual studio 2017.2

@alanpurple why delete the template that was carefully crafted to help you come up with a good bug report, and replace it by a really hard to understand, incomplete account of stuff that maybe you can understand but nobody else?

It makes it really frustrating to try helping you.

If you really want others to help you, try to come up with an [MCVE](http://stackoverflow.com/help/mcve) that demonstrates the problem.

**Comment by alanpurple** [comment]
@dscho 
I don't think I need 'help', I think it's bug for everyone. since this occurs recently and I've never seen this before
If you think this is my own problem, short words like "It's your fault' would suffice

I deleted all format because those short error messages are all and enough

anyway I will try to use the format if I have to

**Comment by fourpastmidnight** [comment]
FWIW, at the moment, I use bitbucket at work and I use Git for Windows and I've never had an issue. As @dscho said, please use the template and provide an MCVE.

Given the limited info, I would say your bitbucket URL is incorrect--but until I (or anyone else) see a MCVE, I'm just throwing mud at a wall to see if it'll stick....

**Comment by alanpurple** [comment]
@fourpastmidnight 

For the repository and visual studio setting I've used for long, this doesn't the case, for new settings in new PC this happens, same URL, so this is not a problem of URL, and even if URL was wrong, then shell also should fail

Thanks anyway,

**Comment by dscho** [comment]
> I don't think I need 'help',

Can we close this ticket, then?

:grinning: 

Seriously again, you clearly encountered a problem, and you clearly were not able to resolve it yourself, and you clearly asked for help here.

It is true that others *may* experience the same issue and will be happy to find this ticket and to find any advice how to resolve it.

But to make that really useful for other people, we need to solve it. And to that end, you need to provide a good bug report with enough information.

**Comment by alanpurple** [comment]
@dscho 

```
fatal: UriFormatException encountered.
actualUrl
fatal: could not read Password for 'https://alanpurple@bitbucket.org': terminal prompts disabled
Pushing to https://alanpurple@bitbucket.org/alanpurple/advisordemo.git
```

this is very simple situation and nothing specific
that url is correct since I could push/clone using shell -> git push
seriously, that's all, what else you need?

you can close if you want

**Comment by alanpurple** [comment]
@dscho 
I think I can handle it, it may be caused by some kind of prompt setting

thank you anyway

**Comment by WiteCastle** [comment]
@alanpurple Did you figure this out? I'm having the same issue with a bitbucket repository. 

Also, @dscho here's the information you requested from above:

### Setup

 - Which version of Git for Windows are you using? Is it 32-bit or 64-bit?

```
$ git --version --build-options

git version 2.13.1.windows.2
built from commit: a36e14b3aaa0ddb633c40a9c8483e6fe12d99616
sizeof-long: 4
machine: x86_64

```

 - Which version of Windows are you running? Vista, 7, 8, 10? Is it 32-bit or 64-bit?

```
$ cmd.exe /c ver

Microsoft Windows [Version 6.1.7601]
```

 - What options did you set as part of the installation? Or did you choose the
   defaults?

```
# One of the following:
$ cat "C:\Program Files\Git\etc\install-options.txt"

Path Option: Cmd
SSH Option: OpenSSH
CURL Option: OpenSSL
CRLF Option: CRLFAlways
Bash Terminal Option: MinTTY
Performance Tweaks FSCache: Enabled
Use Credential Manager: Enabled
Enable Symlinks: Disabled
```

 - Any other interesting things about your environment that might be related
   to the issue you're seeing?

I'm getting this error on a clean clone as well. You can try to reproduce it by attempting to clone the repo and not giving a password. Here's an example:

```
$ git clone https://ebsanu@bitbucket.org/ebsanu/kitchensink.git
Cloning into 'kitchensink'...
fatal: UriFormatException encountered.
   actualUrl
```

### Details

 - Which terminal/shell are you running Git from? e.g Bash/CMD/PowerShell/other

I've tried in the Git Bash and CMD.

 - What commands did you run to trigger this issue? If you can provide a

```
$ git clone https://ebsanu@bitbucket.org/ebsanu/kitchensink.git
Cloning into 'kitchensink'...
fatal: UriFormatException encountered.
   actualUrl
```


 - Any more information?

It looks like this was a known issue in version 2.9.x and occurred because of a port number in the URL. But I've checked and my origin address on bitbucket has no port number that I can see.

**Comment by dscho** [comment]
> UriFormatException encountered.

This may be the symptom fixed in Git Credential Manager 1.10.1 that shipped today in Git for Windows v2.13.2.

Could you give that a spin?

**Comment by alanpurple** [comment]
@WiteCastle  after all, this wasn't my fault as @dscho  said

solved by manually installing GCMW 1.10.1

**Comment by WiteCastle** [comment]
@dscho Yes, that solved it. Thank you!
